haystack/preview/components/websearch/serper_dev.py

```diff
@@ -0,0 +1,161 @@
+            search_params=self.search_params,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "SerperDev":
+        """
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        query: str,
+        allowed_domains: Optional[List[str]] = None,
+        top_k: Optional[int] = None,
+        search_params: Optional[Dict[str, Any]] = None,
+    ) -> List[Document]:
+        """
+        Search the SerperDev API for the given query and return the results as a list of Documents.
+
+        :param query: Query string.
+        :param allowed_domains: List of domains to limit the search to.
+        :param top_k: Number of documents to return.
+        :param search_params: Additional parameters passed to the SerperDev API.
+
+        Refer to the [Serper Dev website](https://serper.dev/) for more details on the API parameters.
+        :return: List[Document]
+        """
+        top_k = top_k or self.top_k
+        allowed_domains = allowed_domains or self.allowed_domains
+        search_params = search_params or {}
+        search_params = {**self.search_params, **search_params}
+        query_prepend = "OR ".join(f"site:{domain} " for domain in allowed_domains) if allowed_domains else ""
+
+        url = "https://google.serper.dev/search"
+
+        payload = json.dumps({"q": query_prepend + query, "gl": "us", "hl": "en", "autocorrect": True, **search_params})
+        headers = {"X-API-KEY": self.api_key, "Content-Type": "application/json"}
+
+        try:
+            response = requests.post(url, headers=headers, data=payload, timeout=30)
+            response.raise_for_status()  # Will raise an HTTPError for bad responses
+            # If we reached this point, it means the request was successful and we can proceed
+
+            json_result = response.json()
+
+            # we get the snippet from the json result and put it in the content field of the document
+            organic = [
+                Document.from_dict(
+                    {"metadata": {k: v for k, v in d.items() if k != "snippet"}, "content": d["snippet"]}
+                )
+                for d in json_result["organic"]
+            ]
+
+            answer_box = []
+            # answer box is what search engine shows as a direct answer to the query
+            if "answerBox" in json_result:
+                answer_dict = json_result["answerBox"]
+                highlighted_answers = answer_dict.get("snippetHighlighted")
+                answer_box_content = (
+                    highlighted_answers[0]
+                    if isinstance(highlighted_answers, list) and len(highlighted_answers) > 0
+                    else highlighted_answers
+                    if isinstance(highlighted_answers, str)
+                    else None
+                )
```

Comment:
I would change it to something similar, having it all in a single line is really confusing.

```suggestion
                answer_box_content = None
                if isinstance(highlighted_answers, list) and len(highlighted_answers) > 0:
                    answer_box_content = highlighted_answers[0]
                elif isinstance(highlighted_answers, str):
                    answer_box_content = highlighted_answers
```